Features/Onboarding/Sources/Navigation/ImportWalletNavigationStack.swift

```diff
@@ -19,27 +18,36 @@ public struct ImportWalletNavigationStack: View {
     }
 
     public var body: some View {
-        NavigationStack {
-            ImportWalletTypeScene(model: model)
-                .toolbar {
-                    ToolbarItem(placement: .navigationBarLeading) {
-                        Button(Localized.Common.cancel) {
-                            dismiss()
+        NavigationStack(path: $navigationPath) {
+            Group {
+                switch model.walletService.isAcceptedTerms {
+                case true:
+                    ImportWalletTypeScene(model: model)
+                case false:
+                    AcceptTermsScene(model: AcceptTermsViewModel(
+                        onNext: {
+                            navigationPath.append(Scenes.ImportWalletType())
                         }
-                    }
+                    ))
                 }
-                .navigationBarTitleDisplayMode(.inline)
-                .navigationDestination(for: ImportWalletType.self) { type in
-                    ImportWalletScene(
-                        model: ImportWalletViewModel(
-                            type: type,
-                            walletService: model.walletService,
-                            onFinishImport: {
-                                isPresentingWallets.toggle()
-                            }
-                        )
+            }
+            .toolbarDismissItem(title: .cancel, placement: .topBarLeading)
+            .navigationBarTitleDisplayMode(.inline)
+            .navigationDestination(for: ImportWalletType.self) { type in
+                ImportWalletScene(
+                    model: ImportWalletViewModel(
+                        type: type,
+                        walletService: model.walletService,
+                        onFinishImport: {
+                            model.acceptTerms()
+                            isPresentingWallets.toggle()
```

Comment:
```suggestion
                            isPresentingWallets = false
```

be specific, avoid toggle in such cases